src/Orchard/Environment/Extensions/ExtensionManager.cs

```diff
@@ -107,14 +107,16 @@ public class ExtensionManager : IExtensionManager {
             Logger.Information("Loading features");
 
             // generate a cachekey by hashing the ids of all feature descriptors
-            var cacheKey = BitConverter.ToString(
-                _md5.ComputeHash(
-                    Encoding.UTF8.GetBytes(
-                        string.Join(";",
-                            featureDescriptors
-                                .Select(fd => fd.Id)
-                                .OrderBy(x => x)))));
-
+            string cacheKey;
+            lock (_md5) {
```

Comment:
That could work but why not just create a custom instance every time and dispose it?

But if the issue happens it's because LoadFeatures is called concurrently, and that could limit the usage of the cache (cache stampede). So maybe create a new MD5 and also prevent multiple results for the same key from being created at the same time using this thing: https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/OutputCaching/src/DispatcherExtensions.cs#L8

If that's too complex then feel free to merge what you just did.